e2e/cypress/integration/api/iam/11_projects_api_action_ingestion.spec.ts

```diff
@@ -0,0 +1,161 @@
+        conditions: [
+          {
+            attribute: 'CHEF_ORGANIZATION',
+            operator: 'EQUALS',
+            values: ['Team Pizza']
+          }
+        ]
+      }
+    },
+    {
+      project: {
+        id: `${cypressPrefix}-project-chef-server-${Cypress.moment().format('MMDDYYhhmm')}`,
+        name: 'project chef server'
+      },
+      rule: {
+        id: 'rule-chef-server',
+        name: 'Chef Server',
+        type: 'EVENT',
+        project_id: `${cypressPrefix}-project-chef-server-${Cypress.moment().format('MMDDYYhhmm')}`,
+        conditions: [
+          {
+            attribute: 'CHEF_SERVER',
+            operator: 'EQUALS',
+            values: ['example.pizza']
+          }
+        ]
+      }
+    }
+  ];
+
+  // we run most of the API calls, including the time-intensive apply-rules,
+  // in the before block so we only have to run them once
+  before(() => {
+    cy.cleanupV2IAMObjectsByIDPrefixes(cypressPrefix, ['projects', 'policies']);
+
+    // create the projects with one rule
+    projectsWithRule.forEach(projectWithRule => {
+      cy.request({
+        headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+        method: 'POST',
+        url: '/apis/iam/v2beta/projects',
+        body: projectWithRule.project
+      });
+
+      cy.request({
+        headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+        method: 'POST',
+        url: `/apis/iam/v2beta/projects/${projectWithRule.rule.project_id}/rules`,
+        body: projectWithRule.rule
+      });
+    });
+
+    // apply rules
+    cy.request({
+      headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+      method: 'POST',
+      url: '/apis/iam/v2beta/apply-rules'
+    });
+
+    cy.waitUntilApplyRulesNotRunning(100);
+
+    // Ingest an action with attributes that match all the projects
+    cy.fixture('action/environment_create.json').then((action) => {
+      action.organization_name = 'Team Pizza';
+      action.remote_hostname = 'example.pizza';
+      action.id = actionId;
+      action.entity_name = entityName;
+      action.recorded_at = Cypress.moment().utc().subtract(1, 'day').format();
+
+      cy.request({
+        headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+        method: 'POST',
+        url: '/data-collector/v0',
+        body: action
+      });
+    });
+
+    // wait for the action to be ingested
+    waitForAction(entityName, 30);
+  });
+
+  after(() => cy.cleanupV2IAMObjectsByIDPrefixes(cypressPrefix, ['projects', 'policies']));
+
+  projectsWithRule.forEach((projectWithRule) => {
+    const attribute = projectWithRule.rule.name;
+    it(`when a project has a rule that matches an action's ${attribute},
+    successfully associates that action with the project`, () => {
+      const end = Cypress.moment().utc().endOf('day').valueOf().toString();
+      const start = Cypress.moment().utc().subtract(3, 'day').valueOf().toString();
```

Comment:
What is the significance of 3 days here...?

Reply:
The event is added a day before now. To ensure we are getting the event we are pulling from three days ago until now.